.github/buildomat/jobs/deploy.sh

```diff
@@ -106,8 +109,29 @@ pfexec chown build:build /opt/oxide/work
 cd /opt/oxide/work
 
 ptime -m tar xvzf /input/package/work/package.tar.gz
+mkdir tests
+for p in /input/build-end-to-end-tests/work/*.gz; do
+	ptime -m gunzip < "$p" > "tests/$(basename "${p%.gz}")"
+	chmod a+x "tests/$(basename "${p%.gz}")"
+done
+
 ptime -m pfexec ./tools/create_virtual_hardware.sh
 
+#
+# Image-related tests use images served by catacomb. The lab network is
+# IPv4-only; the propolis zones are IPv6-only. These steps set up tcpproxy
+# configured to proxy to catacomb via port 54321 in the global zone.
+#
+pfexec mkdir -p /usr/oxide
+pfexec rm -f /usr/oxide/tcpproxy
+pfexec curl -sSfL -o /usr/oxide/tcpproxy \
+	http://catacomb.eng.oxide.computer:12346/tcpproxy
+pfexec chmod +x /usr/oxide/tcpproxy
+pfexec rm -f /var/svc/manifest/site/tcpproxy.xml
+pfexec curl -sSfL -o /var/svc/manifest/site/tcpproxy.xml \
```

Comment:
For the moment I'm avoiding this because I'd rather have the tcpproxy binary and its service manifest in sync in the same place. (But I'm making a note to eventually build tcpproxy in CI perhaps?)